### PR TITLE
Location Picker UI

### PR DESCRIPTION
**Pull request by vermayash8**
### Contains
* Place Picker
* SavedPlacesActivity

### CODE REVIEW

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_saved_places);
+        getSupportActionBar().setDisplayShowHomeEnabled(true);
+        // Get the list of saved locations.
+        mTaskRepository = new TaskRepository(getApplicationContext());
+        List<LocationModel> locations = mTaskRepository.getAllLocations();
+        // Sort in descending order of use count.
+        Collections.sort(locations, (o1, o2) -> o2.getUseCount() - o1.getUseCount());
```
Comment (Shilpi75) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java]: we can also retrieve the sorted data. Which approach is better?

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        });
+    }
+
+    /**
+     * Shows the empty view if numLocations == 0
+     *
+     * @param numLocations the number of locations being shown.
+     */
+    private void setEmptyView(int numLocations) {
+        if (numLocations != 0) {
```
Comment (Shilpi75) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java]: Even if numLocations!=0, we should show choose from map button?

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    private TaskRepository mTaskRepository;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_saved_places);
+        getSupportActionBar().setDisplayShowHomeEnabled(true);
+        // Get the list of saved locations.
+        mTaskRepository = new TaskRepository(getApplicationContext());
+        List<LocationModel> locations = mTaskRepository.getAllLocations();
```
Comment (Shilpi75) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java]: Get all locations for which isHidden = 0.

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+        callingIntent.putExtra(EXTRA_LOCATION_ID, location.getId());
+        this.setResult(RESULT_OK, callingIntent);
+        finish();
+    }
+
+
+    /**
+     * RecyclerView Adapter for the showing the list of saved places.
+     */
+    private class SavedPlacesAdapter extends RecyclerView.Adapter<SavedPlacesAdapter
```
Comment (Shilpi75) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java]: Should we this class here only? Not a separate adapter class?

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java (hunk 3 above)
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java]: Resolved. Taken care of in Location Dao query.

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java (hunk 1 above)
Comment (vermayash8) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java]: Don't know about the internal implementation in SQLite. Reasons for me choosing Java's sort:
1. We need all the locations in memory, so sorting doesn't reduce the data size.
2. We might have some other criteria to sort tomorrow, which can be complex and won't look good as an SQLite query.
3. Not sure about that but, as data increases, SQLite requires the table to be indexed to return results faster. Also, considering the fact that normally in-memory sorting will always be faster than having some part in disk(assuming we are able to keep all data in memory).

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java (hunk 2 above)
Comment (vermayash8) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java]: Earlier the text shown was `You don't have any saved places. Go back and pick a place from the map`. This requires the user to go back and then click pick place. So, I've added a button here only. The screenshot has been attached.
screenshot_2018-01-03-21-12-14-787_app tasknearby yashcreations com tasknearby debug

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java (hunk 4 above)
Comment (vermayash8) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java]: will create a new class afterwards.

**Pull request merged**